src/structures/MessageEmbed.js

```diff
@@ -345,15 +345,30 @@ class MessageEmbed {
     return this;
   }
 
+  /**
+   * The options to provide for setting an author for a {@link MessageEmbed}.
+   * @typedef {Object} MessageEmbedAuthor
+   * @property {string} name The name of this author.
+   * @property {string} [url] The URL of this author.
+   * @property {string} [iconURL] The icon URL of this author.
+   */
+
+  // TODO: Remove the deprecated code in the following method and typings.
   /**
    * Sets the author of this embed.
-   * @param {string} name The name of the author
-   * @param {string} [iconURL] The icon URL of the author
-   * @param {string} [url] The URL of the author
+   * @param {string | MessageEmbedAuthor} options The options to provide for the author.
+   * A string may simply be provided if only the author name is desirable.
+   * @param {string} [deprecatedIconURL] The icon URL of this author. This parameter is **deprecated**.
+   * @param {string} [deprecatedURL] The URL of this author. This parameter is **deprecated**.
    * @returns {MessageEmbed}
    */
-  setAuthor(name, iconURL, url) {
-    this.author = { name: Util.verifyString(name, RangeError, 'EMBED_AUTHOR_NAME'), iconURL, url };
+  setAuthor(options = {}, deprecatedIconURL, deprecatedURL) {
+    if (typeof nameOrOptions === 'string') {
```

Comment:
I might be missing something here, but where is `nameOrOptions` defined?

Reply:
Oh no!